animation_nodes/nodes/gpencil/gp_frame_from_strokes.py

```diff
@@ -0,0 +1,49 @@
+import bpy
+import numpy as np
+from bpy.props import *
+from ... events import propertyChanged
+from ... data_structures import GPFrame
+from ... base_types import AnimationNode, VectorizedSocket
+
+class GPFrameFromStrokesNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPFrameFromStrokesNode"
+    bl_label = "GP Frame From Strokes"
+    errorHandlingType = "EXCEPTION"
+
+    useStrokeList: VectorizedSocket.newProperty()
+    useNumberList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput(VectorizedSocket("GPStroke", "useStrokeList",
+            ("Stroke", "stroke"), ("Strokes", "strokes")), dataIsModified = True)
+        self.newInput(VectorizedSocket("Float", "useNumberList",
+            ("Frame Number", "frameNumber"), ("Frame Numbers", "frameNumbers")), dataIsModified = True)
+        self.newOutput(VectorizedSocket("GPFrame", "useNumberList",
+            ("Frame", "frame"), ("Frames", "frames")))
+
+    def getExecutionFunctionName(self):
+        if self.useNumberList:
+            return "execute_StrokesFrameNumbers"
+        else:
+            return "execute_StrokesFrameNumber"
+
+    def execute_StrokesFrameNumber(self, strokes, frameNumber):
+        if not self.useStrokeList:
+            strokes = [strokes]
+
+        return GPFrame(strokes, 0, frameNumber)
+
+    def execute_StrokesFrameNumbers(self, strokes, frameNumbers):
+        frames = []
+        if not self.useStrokeList:
+            return frames
```

Comment:
Note that you can add multiple vectorized socket properties per socket like we did before. This should remove the case where `self.useNumberList` is true and `self.useStrokeList` is not.

Reply:
Omar, I have changed the code and it worked,
```
def create(self):
        self.newInput(VectorizedSocket("GPStroke", "useStrokeList",
            ("Stroke", "stroke"), ("Strokes", "strokes")), dataIsModified = True)
        self.newInput(VectorizedSocket("Float", ["useStrokeList", "useNumberList"],
            ("Frame Number", "frameNumber"), ("Frame Numbers", "frameNumbers")))
        self.newOutput(VectorizedSocket("GPFrame", "useNumberList",
            ("Frame", "frame"), ("Frames", "frames")))
```
However, if I suddenly cut the input (stokes and frame-numbers) then output does not fall back to _Frame_ . However, it remains _Frames_ which cause problem in _GP Layer From Frames_ node give an error,
![Screenshot from 2020-02-01 00-31-31](https://user-images.githubusercontent.com/30294746/73566587-3d217580-448a-11ea-9ed5-516cbd377591.png)
![Screenshot from 2020-02-01 00-30-56](https://user-images.githubusercontent.com/30294746/73566588-3d217580-448a-11ea-9f1e-10d688925bd6.png)

Reply:
If I use this code then the output fall back to _Frame_  and the above error does not occur,
```
def create(self):
        self.newInput(VectorizedSocket("GPStroke", "useStrokeList",
            ("Stroke", "stroke"), ("Strokes", "strokes")), dataIsModified = True)
        self.newInput(VectorizedSocket("Float", ["useStrokeList", "useNumberList"],
            ("Frame Number", "frameNumber"), ("Frame Numbers", "frameNumbers")))
        if self.useNumberList:
            self.newOutput("GPFrame List", "Frames", "frames")
        else:
            self.newOutput("GPFrame", "Frame", "frame")
```

Reply:
I can't replicate the issue. But I will take a look at the vectorization code, see if something is up.